### PR TITLE
[Instrumentation.Process] Dropping Snapshot

### PR DESCRIPTION
**Pull request by Yun-Ting**
Update to use Process.Refresh() on each callback after benchmarking the perf https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/717 and found it cheap.
Also, addressing https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/664#discussion_r993403498.

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs [hunk 1]
```diff
@@ -68,14 +56,8 @@ public ProcessMetrics(ProcessInstrumentationOptions options)
...
             {
-                if (!this.virtualMemoryUsageBytes.HasValue)
-                {
-                    this.Snapshot();
-                }
-
-                var value = this.virtualMemoryUsageBytes.Value;
-                this.virtualMemoryUsageBytes = null;
-                return value;
+                this.currentProcess.Refresh();
```
Comment (reyang) [inline src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs:59]: Do we really need to keep using the same instance and do `Refresh` each time, or just creating a new Process instance is sufficient? Would the `Refresh` call here on the same instance cause race condition?

Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs:59]: Race condition is possible here. I don't know what's the guarantee of the .Refresh() API so if we want to do Refresh on each callback, locking it might be the safest.

Not sure which one is cheaper, refresh + lock or creating a new instance each time. I'll do some experiments.

Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs]: I've benchmarked the above 2 options https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/717#issue-1415674535.
There is no significance difference in time. I updated the code to use the latter option.

**Review by noahfalk — APPROVED** [review]
LGTM!

(I still have concerns about the cpu utilization metric and using PrivateMemory for the process.memory.virtual metric, but neither of those concerns needs to hold up this PR)

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs (hunk 1 above)
Comment (reyang) [reply src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs]: Thanks!

**Review by Kielek — APPROVED** [review]

File: test/OpenTelemetry.Instrumentation.Process.Tests/ProcessMetricsTests.cs [hunk 2]
```diff
@@ -158,7 +158,7 @@ private static double GetValue(Metric metric)
         {
             if (metric.MetricType.IsGauge())
             {
-                sum += metricPoint.GetGaugeLastValueDouble();
+                sum += metricPoint.GetGaugeLastValueLong();
```
Comment (utpilla) [inline test/OpenTelemetry.Instrumentation.Process.Tests/ProcessMetricsTests.cs:161]: Why this change? Also, the if-else condition seems incorrect. Were you trying to do something similar to this: 

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/test/OpenTelemetry.Instrumentation.Runtime.Tests/RuntimeMetricsTests.cs#L187-L198

Comment (Yun-Ting) [reply test/OpenTelemetry.Instrumentation.Process.Tests/ProcessMetricsTests.cs:161]: process.memory.usage is using .NET API WorkingSet64 which is `long`.
Previously, we used `double?` to cache the value. Thus, the change.

I've removed the unnecessary branching.

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Review by utpilla — APPROVED** [review]
